src/redisearch_rs/inverted_index/src/fields_offsets.rs

```diff
@@ -108,9 +117,16 @@ impl Decoder for FieldsOffsets {
             cursor.seek(SeekFrom::Current(offsets_sz as i64))?;
         };
 
-        let record =
-            decode_term_record_offsets(cursor, base, 0, field_mask as t_fieldMask, 1, offsets_sz)?;
-        Ok(Some(record))
+        decode_term_record_offsets(
+            cursor,
+            base,
+            0,
+            field_mask as t_fieldMask,
+            1,
+            offsets_sz,
+            result,
+        )?;
+        Ok(true)
```

Comment:
Can it be simplified to return the result of `decode_term_record_offsets`?

Edit: No, `decode_term_record_offsets` returns `std::io::Result<()>`

Reply:
Yeah, the return types are different and it does not make sense to return a `bool` from `decode_term_record_offsets`